File: app/src/main/java/www/dico/cn/partybuild/activity/LoginActivity.java
package www.dico.cn.partybuild.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;

import com.google.gson.Gson;

import butterknife.BindView;
import butterknife.ButterKnife;
import www.dico.cn.partybuild.AppConfig;
import www.dico.cn.partybuild.MainActivity;
import www.dico.cn.partybuild.R;
import www.dico.cn.partybuild.bean.LoginBean;
import www.dico.cn.partybuild.modleview.LoginView;
import www.dico.cn.partybuild.mvp.factory.CreatePresenter;
import www.dico.cn.partybuild.mvp.view.AbstractMvpActivity;
import www.dico.cn.partybuild.presenter.LoginPresenter;

//登录
@CreatePresenter(LoginPresenter.class)
public class LoginActivity extends AbstractMvpActivity<LoginView, LoginPresenter> implements LoginView {
    @BindView(R.id.et_name_login)
    EditText et_name_login;
    @BindView(R.id.et_password_login)
    EditText et_password_login;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        ButterKnife.bind(this);
//        Typeface typeface = Typeface.createFromAsset(getAssets(), "fonts/fangzheng.ttf");
//        tv_login_ok.setTypeface(Typeface.defaultFromStyle(Typeface.BOLD_ITALIC));
    }

    public void login(View view) {
        String login_name = et_name_login.getText().toString().trim();
        String password = et_password_login.getText().toString().trim();
        if (TextUtils.isEmpty(login_name)) {
            showToast("用户名不能为空");
        } else if (TextUtils.isEmpty(password)) {
            showToast("密码不能为空");
        } else {
            getMvpPresenter().clickRequest(et_name_login.getText().toString().trim(), et_password_login.getText().toString().trim());
        }
    }

    @Override
    public void resultSuccess(String result) {
        LoginBean bean = new Gson().fromJson(result, LoginBean.class);
        if (bean.getCode().equals("0000")) {
            if (null != bean.getData()) {
                String partyPosition = bean.getData().getPartyBranchPost();//党内职务
                String position = bean.getData().getPosition();//行政职务
                String token = bean.getData().getToken();
                String userId = bean.getData().getUserId();//用户id
                String avatar = bean.getData().getAvatar();//头像
                boolean isManager = bean.getData().isManager();//是否为管理员
                partyPosition = (null == partyPosition) ? "" : partyPosition;
                AppConfig.getSpUtils().put("partyBranchPost", partyPosition);
                position = (null == position) ? "" : position;
                AppConfig.getSpUtils().put("position", position);
                AppConfig.getSpUtils().put("token", "Bearer " + token);
                AppConfig.getSpUtils().put("userId", userId);
                AppConfig.getSpUtils().put("avatar", avatar);
                AppConfig.getSpUtils().put("isManager", isManager);
                AppConfig.getSpUtils().put("isLoginOk",1);
                goTo(MainActivity.class, null);
                finish();
            }
        } else {
            showToast(bean.getMsg());
        }
    }

    @Override
    public void resultFailure(String result) {
        showToast(result);
    }

    @Override
    public void netWorkUnAvailable() {
        showToast("网络出现异常");
    }
}

File: app/src/main/java/www/dico/cn/partybuild/activity/LoginActivity.kt
package www.dico.cn.partybuild.activity

import android.os.Bundle
import android.text.Editable
import android.text.TextUtils
import android.view.View
import com.google.gson.Gson
import kotlinx.android.synthetic.main.activity_login.*
import www.dico.cn.partybuild.AppConfig
import www.dico.cn.partybuild.MainActivity
import www.dico.cn.partybuild.R
import www.dico.cn.partybuild.bean.LoginBean
import www.dico.cn.partybuild.modleview.LoginView
import www.dico.cn.partybuild.mvp.factory.CreatePresenter
import www.dico.cn.partybuild.mvp.view.AbstractMvpActivity
import www.dico.cn.partybuild.presenter.LoginPresenter
import www.dico.cn.partybuild.utils.SPUtils

//登录
@CreatePresenter(LoginPresenter::class)
class LoginActivity : AbstractMvpActivity<LoginView, LoginPresenter>(), LoginView {
    private var username = ""
    private val sp: SPUtils = AppConfig.getSpUtils()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        cb_keep_pwd.isChecked = false
        if (sp.getBoolean("isKeep")) {
            et_name_login.text = Editable.Factory.getInstance().newEditable(sp.getString("username"))
            et_password_login.text = Editable.Factory.getInstance().newEditable(sp.getString("password"))
        }
    }

    fun login(view: View) {
        username = et_name_login!!.text.toString().trim { it <= ' ' }
        val password = et_password_login!!.text.toString().trim { it <= ' ' }
        if (TextUtils.isEmpty(username)) {
            showToast("用户名不能为空")
        } else if (TextUtils.isEmpty(password)) {
            showToast("密码不能为空")
        } else {
            if (cb_keep_pwd.isChecked) {
                sp.put("isKeep", true)
            } else {
                sp.put("isKeep", false)
            }
            sp.put("username", username)
            sp.put("password", password)
            mvpPresenter.clickRequest(dialog, et_name_login!!.text.toString().trim { it <= ' ' }, et_password_login!!.text.toString().trim { it <= ' ' })
        }
    }

    override fun resultSuccess(result: String) {
        val bean = Gson().fromJson<LoginBean>(result, LoginBean::class.java)
        if (bean.code == "0000") {
            if (null != bean.data) {
                var name: String? = bean.data.name
                var partyPost: String? = bean.data.partyPost//党内职务
                val token = bean.data.token
                val userId = bean.data.userId//用户id
                val avatar = bean.data.avatar//头像
                val isManager = bean.data.IsManager()//是否为管理员

                name = if (null == name) "" else bean.data.name
                sp.put("name", name)
                partyPost = if (null == partyPost) "" else bean.data.partyPost
                sp.put("partyPost", partyPost)
                sp.put("token", "Bearer $token")
                sp.put("userId", userId)
                sp.put("avatar", avatar)
                sp.put("isManager", isManager)
                sp.put("isLoginOk", 1)
                goTo(MainActivity::class.java, null)
                finish()
            }
        } else {
            showToast(bean.msg)
        }
    }

    override fun resultFailure(result: String) {
        showToast(result)
    }

    override fun netWorkUnAvailable() {
        showToast("网络出现异常")
    }
}
